webapp/src/app.tsx

```diff
@@ -4592,7 +4592,7 @@ document.addEventListener("DOMContentLoaded", () => {
             const cloudLang = auth.getState()?.preferences?.language;
             // kick of a user preferences check; if the language is different we'll reload
             auth.initialUserPreferences().then((pref) => {
-                if (pref && pref.language !== pxt.BrowserUtils.getCookieLang()) {
+                if (pref && pref.language != pxt.BrowserUtils.getCookieLang()) {
                     pxt.BrowserUtils.setCookieLang(pref.language);
```

Comment:
if the user does not allow you to set cookies for any kind of reason, you reload forever.

Reply:
Good point.

We need to purge this reload and be non-cookie safe.

Reply:
I removed the reload entirely and put a info message. I also added a check to see if language cookie exists at all. This branch should only be hit if you change preferences on a different computer.